plugins/source/src/main/kotlin/app/aaps/plugins/source/XdripSourcePlugin.kt

```diff
@@ -106,7 +106,7 @@ class XdripSourcePlugin @Inject constructor(
                 raw = round(bundle.getDouble(Intents.EXTRA_RAW, 0.0)),
                 noise = null,
                 trendArrow = TrendArrow.fromString(bundle.getString(Intents.EXTRA_BG_SLOPE_NAME)),
-                sourceSensor = SourceSensor.fromString(bundle.getString(Intents.XDRIP_DATA_SOURCE_DESCRIPTION) ?: "")
+                sourceSensor = SourceSensor.fromString(bundle.getString(Intents.XDRIP_DATA_SOURCE) ?: "")
```

Comment:
Based on your feedback I would first check if XDRIP_DATA_SOURCE_DESCRIPTION is present, if not then fallback to XDRIP_DATA_SOURCE. That way behavior for other sources will stay the same, and also work for G7 ?

Reply:
XDRIP_DATA_SOURCE_DESCRIPTION is present in every xDrip BgBroadcast. However it may contain multiple sensors, which is not ideal. I traced the paths for both direct and companion app BGcollectors in xDrip. I don't think the complexity of checking both sourceDesc and source_info fields is necessary. Here are some reasons why I think checking source_info is a better solution.

1. xDrip sends BG with 'source_info' to nightscout.
Both collectors create BGReadings which are later used to send BgBroadcasts.
The source_info is stored within each BGReading when it is created by both direct and companion/UI collectors as follows:

| sensor | source_info |
|--------|--------|
| G7 | G7 Native|
| G6 | G6 Native|
| G5 | G5 Native| 

2. xDrip performs translations on sourceDesc, specifically for enabling SMBAlways to satisfy the AAPS sensor 'advancedFiltering' verification.
This translation was initially introduced when G6 support was added to make a G6 look like a G5 to AAPS to avoid updating AAPS.
https://github.com/NightscoutFoundation/xDrip/commit/07184994b025e7ccc46bdd8c246bb4190a785133?diff=split&w=0
When xDrip sends BGBroadcasts, the sourcDesc field is populated by translating the BGcollector HardwareName as follows:

| sensor | direct BGcollector sourceDesc | UIcollector sourceDesc|
|--------|--------|--------|
| G7 | **G6 Native/ G7 / G5 Native** |  "UI Collector" |
| G6 | G6 Native / G5 Native |  "UI Collector" |
| G5 | G5 Native |  "UI Collector" |

3. The current approach of xDrip translating sourceDesc is not maintainable.
When a G9 is released, using the current approach the sourceDesc would be:
 'G6 Native / G7 / G8? / G9? / G5 Native'
 This adds unnecessary complexity to both xDrip and AAPS.

IMO I don't think AAPS should be accepting G7 sensor data as a G6 or any other sensor. xDrip should be responsible for simply collecting and broadcasting BG.

Reply:
Valid, I will test with G6 and let you know